frontend/apps/reader/modules/readerwikipedia.lua

```diff
@@ -59,6 +61,146 @@ function ReaderWikipedia:addToMainMenu(menu_items)
+                                    text = _("Cancel"),
+                                    callback = function()
+                                        UIManager:close(wikilang_input)
+                                    end,
+                                },
+                                {
+                                    text = _("Save"),
+                                    is_enter_default = true,
+                                    callback = save_wikilang,
+                                },
+                            }
+                        },
+                    }
+                    wikilang_input:onShowKeyboard()
+                    UIManager:show(wikilang_input)
+                end,
+            },
+            { -- setting used by dictquicklookup
+                text = _("Set Wikipedia 'Save as EPUB' directory"),
+                callback = function()
+                    local folder_path_input
+                    local function save_folder_path()
+                        local folder_path = folder_path_input:getInputText()
+                        folder_path = folder_path:gsub("^%s*(.-)%s*$", "%1") -- trim spaces
+                        folder_path = folder_path:gsub("^(.-)/*$", "%1") -- remove trailing slash
+                        if folder_path == "" then
+                            G_reader_settings:delSetting("wikipedia_save_dir", folder_path)
+                        else
+                            if lfs.attributes(folder_path, "mode") == "directory" then -- existing directory
+                                G_reader_settings:saveSetting("wikipedia_save_dir", folder_path)
+                            elseif lfs.attributes(folder_path) then -- already exists, but not a directory
+                                UIManager:show(InfoMessage:new{
+                                    text = _("A path with that name already exists, but is not a directory.")
+                                })
+                                return
+                            else -- non-existing path, we may create it
+                                local parent_dir, sub_dir = util.splitFilePathName(folder_path) -- luacheck: no unused
+                                if lfs.attributes(parent_dir, "mode") == "directory" then -- existing directory
+                                    lfs.mkdir(folder_path)
+                                    if lfs.attributes(folder_path, "mode") == "directory" then -- existing directory
+                                        G_reader_settings:saveSetting("wikipedia_save_dir", folder_path)
+                                        UIManager:show(InfoMessage:new{
+                                            text = _("Directory created."),
+                                        })
+                                    else
+                                        UIManager:show(InfoMessage:new{
+                                            text = _("Failed creating directory.")
```

Comment:
creating directory failed (or "failed *at* creating directory" would also be grammatically correct, though stylistically somewhat awkward)